### PR TITLE
Add Gateway Resource

### PR DESCRIPTION
**Pull request by jonathan34c**
### Which issue this PR addresses:

Add Gateway and Virtual Service, secret resources to poc cluster and namespace. 
### What this PR does / why we need it:

Pack binary in helm in template, run by pipeline to add the gateway/virtual service/ secret resource. 

### Test plan for issue:
Check if the resources has been created .
Screenshot 2023-08-22 144452

### Future work.
1. Once sample app been uploaded, need to point gateway/virtual service to the app service

### CODE REVIEW

File: pkg-0.1.0.tgz
Comment (niontive) [inline pkg-0.1.0.tgz:1]: Please delete this

Comment (jonathan34c) [reply pkg-0.1.0.tgz:1]: oh i forgot to delete this. I accidentally modified this file

File: poc/pkg/values.yaml [hunk 1]
```diff
@@ -80,3 +80,20 @@ nodeSelector: {}
 tolerations: []
 
 affinity: {}
+
+# Use for Istio service mesh 
+istio:
+  name: aks-istio-ingressgateway-internal
```
Comment (niontive) [inline poc/pkg/values.yaml]: This should be `aks-istio-ingressgateway-external`

File: poc/pkg/templates/istio-gateway.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+spec:
+  selector:
+    istio: {{ .Values.istio.name }}
+  servers:
+  - port:
+      number: 443
+      name: http
+      protocol: HTTPS
+    tls:
+      mode: SIMPLE
```
Comment (niontive) [inline poc/pkg/templates/istio-gateway.yaml]: For now, could we disable HTTPS so we can test without this secret?

Comment (jonathan34c) [reply poc/pkg/templates/istio-gateway.yaml]: I have changed the port from 443 to 80, we should able communicate without TLS now.

File: poc/pkg/values.yaml [hunk 3]
```diff
@@ -80,3 +80,22 @@ nodeSelector: {}
...
+    #TODO jonachang: need to specify the host once we have the domain name
+    host: "*"
+    #TODO jonachang: change back to 443 when we have TLS
+    port: 80
+  virtualService:
+    name: rp-poc-virtualservice
+    port: 80
+    #TODO jonachang: need to update to the sample app name once created
+    service: aro-rp-poc-pkg
+  credential:
```
Comment (niontive) [inline poc/pkg/values.yaml]: Helm lint (https://helm.sh/docs/helm/helm_lint/) through an error here. I would just (1) remove the `credential` part and (2) add your comment to line 84/85.

**Pull request merged**